### PR TITLE
Regression: Fix version check for MesonMain methods

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

**Comment by xclaesse** [comment]
Ideally we should make the subproject argument mandatory in `InterpreterObject.__init__()`. I'm not doing that here because this PR should be backported to 0.59.1, so better keep it safe.

File: mesonbuild/interpreterbase/baseobjects.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def __repr__(self) -> str:
         return f'<[{type(self).__name__}] holds [{type(self.held_object).__name__}]: {self.held_object!r}>'
 
 class RangeHolder(MesonInterpreterObject):
-    def __init__(self, start: int, stop: int, step: int, *, subproject: T.Optional[str] = None) -> None:
+    def __init__(self, start: int, stop: int, step: int, subproject: str) -> None:
```
Comment (mensinda) [inline mesonbuild/interpreterbase/baseobjects.py:85]: `subproject` should stay kwarg only (to be consistent with the other calls) but be non-optional

Comment (xclaesse) [reply mesonbuild/interpreterbase/baseobjects.py]: right, done.

**Review by mensinda — APPROVED** [review]
LGTM if the CI also agrees,

**Pull request merged**